build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -40,17 +40,25 @@ class BuildAssetUriResolver extends UriResolver {
   /// When actions can run out of order an asset can move from being readable
   /// (in the later phase) to being unreadable (in the earlier phase which ran
   /// later). If this happens we don't want to hide the asset from the analyzer.
-  final seenAssets = HashSet<AssetId>();
+  final globallySeenAssets = HashSet<AssetId>();
+
+  /// The assets which have been resolved from a [BuildStep], either as an
+  /// input, subsequent calls to a resolver, or a transitive import thereof.
+  final _buildStepAssets = <BuildStep, HashSet<AssetId>>{};
 
   /// Crawl the transitive imports from [entryPoints] and ensure that the
   /// content of each asset is updated in [resourceProvider] and [driver].
   Future<void> performResolve(BuildStep buildStep, List<AssetId> entryPoints,
       AnalysisDriver driver) async {
-    final changedPaths =
-        await crawlAsync<AssetId, _AssetState>(entryPoints, (id) async {
+    final seenInBuildStep =
+        _buildStepAssets.putIfAbsent(buildStep, () => HashSet());
+    final notCrawled = (AssetId asset) => !seenInBuildStep.contains(asset);
```

Comment:
https://dart.dev/guides/language/effective-dart/usage#do-use-a-function-declaration-to-bind-a-function-to-a-name

```suggestion
bool notCrawled(AssetId asset) => !seenInBuildStep.contains(asset);
```